store/tikv/txn.go

```diff
@@ -279,7 +286,36 @@ func (txn *tikvTxn) Commit(ctx context.Context) error {
 	})
 
 	start := time.Now()
-	defer func() { tikvTxnCmdHistogramWithCommit.Observe(time.Since(start).Seconds()) }()
+	if len(txn.lockKeys) > 0 && txn.committer != nil && txn.committer.connID > 0 && txn.IsPessimistic() {
+		startCommitTime := time.Now()
+		txn.commitStartTime = &startCommitTime
+		if txn.lockAcquiredTime != nil {
+			logutil.Logger(ctx).Info("[pessimistic lock keys] txn try to commit",
+				zap.Uint64("start_ts", txn.startTS),
+				zap.Stringer("lock_time", txn.lockAcquiredTime),
+				zap.Stringer("start_commit_time", txn.commitStartTime),
+				zap.Int64("key_lock_time_to_commit_start[in ms]", txn.commitStartTime.Sub(*txn.lockAcquiredTime).Milliseconds()))
+		}
+	}
+
+	var err error
+	defer func() {
+		tikvTxnCmdHistogramWithCommit.Observe(time.Since(start).Seconds())
+		lockReleaseTime := time.Now()
+		txn.lockReleasedTime = &lockReleaseTime
+		if len(txn.lockKeys) > 0 && txn.committer != nil && txn.committer.connID > 0 && txn.IsPessimistic() &&
```

Comment:
@cfzjywxk I prefer removing the log to avoid too many logs, and I added a simple one with forUpdateTS only, see this PR's description.

Reply:
This log is costy, could we print it using the same threshold?

Reply:
Besides, printing all keys `zap.Strings("locked keys", HexKeys(txn.lockKeys)))`, this could be removed.

Reply:
I removed the `HexKeys(keys)` and added a threshold of acquiring the lock.